### ISSUE TITLE
Certificate chain in the same order as it appears in TLS

### ISSUE THREAD

**Issue by martinthomson** [httpwg/http-extensions#1927]
There's a bunch of text in the draft about putting the certificate chain in the right order.  This tries to crib from TLS and ends up being somewhat tricky to get right as TLS has a bunch of carefully worded and - frankly - ambiguous rules.  As a simple alternative, how about we just state that the certificates need to appear in the same order as the client sent them in TLS.

**Comment by bc-pi** [comment]
Makes sense. Will look at adjusting the text accordingly.

**Comment by martinthomson** [comment]
It is definitely 1. You do 2 and you don't need this client-cert-chain field at all.

That doesn't mean that a proxy can't be configured to do the work and pass the info along. But one of the two needs to be in the hook for enforcement.

A consequence of 1 is that alerts at the TLS layer won't be used to signal errors, aside from those that the proxy detects (or has to, like if the signature is invalid). That is ok, mostly. The main protocol issue here that might be a problem is validation of signature_scheme_cert. I can't see how failing to validate that would be a significant problem though.

**Comment by bc-pi** [comment]
In [this conversation](https://github.com/httpwg/http-extensions/pull/1874#pullrequestreview-841081689) on the PR that introduced the chain I was aiming for 2. But eventually relented to MT. Some additional considerations about the security implications are likely warranted though.

**Comment by davidben** [comment]
I see. If it's (1), that should probably be clearer in the draft. That also means the draft needs to define an error-signaling scheme. Moreover, if the proxy doesn't then translate that error into TLS alerts, we'd need changes in clients for servers relying on this to work right. I'll file a bug.

**Comment by b---c** [comment]
I ran into real reservations as I looked at making this change. I still don't like pushing the trust chain validation to the origin. For many reasons. The proxy also has no means of signaling to the client which trust anchors the origin is okay with.  
The current text is somewhat ambiguous about what goes in Client-Cert-Chain but I think that's actually better than saying it has to be same as in the TLS handshake.  I dunno, I think I might raise the question of WTF actually goes in Client-Cert-Chain at the next interim or real meeting.
